### PR TITLE
[1LP][RFR]Ssui console tests

### PR DESCRIPTION
**Pull request by kedark3**
## Purpose or Intent
**Fixing Console** Tests - SSUI
This is built on top of #9595, so we need to merge that first and then merge this.

Passed in jenkins at
5.11 -- job/kk-console-chrome/55/console
5.10 -- job/kk-console-chrome/62/console

### PRT Run
{{pytest: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py -vvv --use-provider complete }}

### CODE REVIEW

**Comment by kedark3** [comment]
Passed in **jenkins** at 
`5.11` -- job/kk-console-chrome/55/console

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 01c151b7f6f6a146d06f9fbddb3d6b77795865a6

The *global* fixture **`order_service`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py**
- **cfme/tests/ssui/test_ssui_dashboard.py**
    - *test_total_services*
- **cfme/tests/ssui/test_ssui_myservice.py**
    - *test_myservice_crud*
- **cfme/tests/ssui/test_ssui_service_catalogs.py**

The *global* fixture **`configure_websocket`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_html5_vm_console.py**
    - *test_html5_vm_console*
- **cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py**
- **cfme/tests/infrastructure/test_webmks_console.py**
    - *test_webmks_vm_console*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 3b34bd4d56207c9615668a568ed910bb817ec05d

The *global* fixture **`order_service`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py**
- **cfme/tests/ssui/test_ssui_dashboard.py**
    - *test_total_services*
- **cfme/tests/ssui/test_ssui_myservice.py**
    - *test_myservice_crud*
- **cfme/tests/ssui/test_ssui_service_catalogs.py**

The *global* fixture **`configure_websocket`** is used in the following files:
- **cfme/tests/cloud_infra_common/test_html5_vm_console.py**
    - *test_html5_vm_console*
- **cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py**
- **cfme/tests/infrastructure/test_webmks_console.py**
    - *test_webmks_vm_console*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/cloud_infra_common/test_html5_vm_console.py [hunk 1]
```diff
@@ -130,6 +130,7 @@ def test_html5_vm_console(appliance, setup_provider, provider, configure_websock
 @pytest.mark.tier(2)
 @pytest.mark.parametrize('browser', ['chrome_latest', 'firefox_latest'])
 @pytest.mark.parametrize('operating_system', ['fedora_latest', 'rhel8.x', 'rhel7.x', 'rhel6.x'])
+@pytest.mark.provider([CloudProvider, InfraProvider], required_flags=['html5_console'])
```
Comment (kedark3) [inline cfme/tests/cloud_infra_common/test_html5_vm_console.py]: Missed these in previous PR #9595

File: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+def test_vm_console_ssui(request, appliance, setup_provider, provider, context,
+     configure_console_webmks, configure_console_vnc, order_service, take_screenshot,
+     configure_websocket, console_template):
+    """Test Myservice VM Console in SSUI.
+
+    Metadata:
+        test_flag: ssui
+
+    Polarion:
+        assignee: apagac
```
Comment (john-dupuy) [inline cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py]: Did you ever hear from Sudhir about this? I don't think we should leave tests assigne to `apagac`

File: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            vm_obj.mgmt.assign_floating_ip(public_net)
+        request.addfinalizer(vm_console.close_console_window)
+        request.addfinalizer(appliance.server.logout)
+        try:
+            assert vm_console.wait_for_connect(180), ("VM Console did not reach 'connected'"
+                " state")
+            # wait for screen text to return non-empty string, which implies console is loaded
+            # and has readable text
+            wait_for(func=lambda: vm_console.get_screen_text() != '', delay=5, timeout=45)
+            assert vm_console.get_screen_text() != '', "VM Console screen text returned Empty"
```
Comment (john-dupuy) [inline cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py]: **Required**
```suggestion
            assert vm_console.get_screen_text(), "VM Console screen text returned Empty"
```

File: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py (hunk 2 above)
Comment (kedark3) [reply cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py]: Nope :(

File: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
+"""Test for HTML5 Remote Consoles of VMware/RHEV/RHOSP Providers."""
+import pytest
+from wait_for import wait_for
```
Comment (mshriver) [inline cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py]: https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/wait.py

Please import wait_for through cfme.utils, so the logger context is passed.

File: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py (hunk 2 above)
Comment (mshriver) [reply cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py]: I'm okay with it, easy to bulk re-assign.

File: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py (hunk 4 above)
Comment (kedark3) [reply cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py]: done

File: cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        test_flag: ssui
+
+    Polarion:
+        assignee: apagac
+        casecomponent: Infra
+        caseimportance: medium
+        initialEstimate: 1/2h
+    """
+    if provider.one_of(VMwareProvider) and appliance.version < '5.11':
+        appliance.server.settings.update_vmware_console({'console_type': 'VNC'})
```
Comment (jawatts) [inline cfme/tests/cloud_infra_common/test_html5_vm_console_ssui.py]: **Question** Should there be a finalizer here to reset it back to the default?

**Pull request merged**